x/campaign/types/voucher.go

```diff
@@ -0,0 +1,56 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+const (
+	// VoucherPrefix is the prefix used to represent a voucher denomination
+	VoucherPrefix = "v/"
+)
+
+// SharesToVouchers returns new Coins vouchers from the Shares representation
+func SharesToVouchers(campaignID uint64, shares Shares) sdk.Coins {
```

Comment:
For both conversion methods, I think we should include a verification that shares are valid shares and same for vouchers and return errors.

We can then use critical errors if shares in the store are invalid shares